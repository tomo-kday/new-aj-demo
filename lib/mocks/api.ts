import { mockData } from "./mockData";

// eslint-disable-next-line no-undef
export const getPlans = jest.fn(() => Promise.resolve(mockData));
