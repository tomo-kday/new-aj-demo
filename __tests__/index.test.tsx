/* eslint-disable no-undef */
import Home from "@/app/page";
import { getPlans } from "@/lib/mocks/api";
import { mockData } from "@/lib/mocks/mockData";

import { render } from "@testing-library/react";
import { act } from "react-dom/test-utils";

describe("Home", () => {
	beforeEach(() => {
		// Mock the API function's implementation to return the mock data
		getPlans.mockResolvedValue(mockData);
	});
	// it('list 30 plans', async () => {
	//   // await act(async () => {
	//   //   const { queryAllByTestId, container } = render(await Home());
	//   //   const plans = await queryAllByTestId('plan-card');
	//   //   await expect(plans).toHaveLength(30);
	//   // });

	//   // await act(async () => {
	//   //   render(await Home());
	//   //   const plans = screen.getAllByTestId('plan-card');
	//   //   expect(plans).toHaveLength(30);
	//   // });
	// });
	it("renders a heading", async () => {
		await act(async () => {
			render(await Home());
		});
	});
	// it('snapshot', async () => {
	//   await act(async () => {
	//     const { container } = await render(<Home />);
	//     expect(container).toMatchSnapshot();
	//   });
	// });
});
