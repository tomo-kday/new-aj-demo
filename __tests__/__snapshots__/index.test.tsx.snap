// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home list 30 plans 1`] = `<div />`;

exports[`Home snapshot 1`] = `<div />`;
